apps/roam/src/components/DiscourseContextOverlay.tsx

```diff
@@ -138,29 +152,88 @@ const DiscourseContextOverlay = ({
     getInfo();
   }, [refresh, getInfo]);
 
-  // Suggestive Mode
-  const validTypes = useMemo(() => {
-    if (!discourseNode) return [];
-    const selfType = discourseNode.type;
-    const validRelations = relations.filter((relation) =>
-      [relation.source, relation.destination].includes(selfType),
+  const suggestionContextData = useMemo(() => {
+    const selfNode = discourseNode;
+
+    if (!selfNode || !selfNode.text || !selfNode.format) {
+      return {
+        validTypes: [],
+        uniqueRelationTypeTriplets: [],
+      };
+    }
+    const selfType = selfNode.type;
+
+    const relationsConnectingToSelf = relations.filter(
```

Comment:
If you rename/refactor a variable/function, but don't change it's purpose, make sure to leave a comment as to why this change was necessary (just in GitHub in the PR, you can click on the `+` to leave a comment on this specific line).

This helps the review understand functionally irrelevant changes much quicker.